grr/server/grr_response_server/gui/ui/components/client_add_label_dialog/client_add_label_dialog.ng.html

```diff
@@ -0,0 +1,24 @@
+<div mat-dialog-title class="title">Adding a label</div>
+<div mat-dialog-content class="content">
+  <p>You can add a new label or select from existing labels.</p>
+  <mat-form-field class="label-field" appearance="outline">
+    <mat-label>Start typing label name</mat-label>
+    <input matInput
+      cdkFocusInitial
+      (keyup.enter)="onAddClick()"
+      [formControl]="newLabelInputControl"
+      [matAutocomplete]="auto">
+
+    <mat-autocomplete #auto="matAutocomplete">
+      <mat-option *ngFor="let option of filteredCurrentLabels | async" [value]="option">
+        {{option}}
+      </mat-option>
+    </mat-autocomplete>
+
+    <mat-error *ngIf="newLabelInputControl.invalid">This label is already present!</mat-error>
+  </mat-form-field>
+</div>
+<div mat-dialog-actions class="actions">
+  <button mat-button color="primary" (click)="onCancelClick()">Cancel</button>
+  <button mat-button color="primary" (click)="onAddClick()">Add label</button>
```

Comment:
Since the onclick-handler does nothing when the form is invalid, this button should be disabled when the form is disabled. Since adding a label can be regarded as an idempotent action (the collection of client labels has the same state, even if you add a label twice), I'd also be fine with not handling this case at all and allowing the user to try to add a label twice.

Reply:
I think that disabling the button would be a nice touch!